spec/services/update_auction_spec.rb

```diff
@@ -2,6 +2,43 @@
 
 describe UpdateAuction do
   describe '#perform' do
+    context 'when publishing an auction' do
+      it 'creates an AuctionState record' do
+        auction = create(:auction, :unpublished)
+
+        params = {
+          auction: {"published" => "published"}
+        }
+
+        expect do
+          UpdateAuction.new(
+            auction: auction,
+            params: params,
+            current_user: auction.user
+          ).perform
+        end.to change { AuctionState.count }.by(1)
+
+        auction.reload
+
+        # do we want to low-level test for the published state object, like this?
+        published_state = auction.states.find {|state| state.name == 'published'}
+        expect(published_state.state_value).to eq('published')
+
+        # and/or this?:
+        # expect(auction.published?).to be true
+
+        # and/or this?:
+        # published = AuctionPublishedState.new(auction).perform
+        # expect(published).to be true
+
```

Comment:
I would only test this as the return value if you already had plans to use it somewhere. Otherwise the maintenance is overhead.